### PR TITLE
BLE patches

### PR DESCRIPTION
**Pull request by jamesadevine**

### CODE REVIEW

**Review by pelikhan — APPROVED** [review]

File: src/jdom/bluetooth.ts [hunk 1]
```diff
@@ -53,7 +55,8 @@ class BluetoothTransport extends JDTransport {
     private _device: BluetoothDevice
     private _server: BluetoothRemoteGATTServer
     private _service: BluetoothRemoteGATTService
-    private _characteristic: BluetoothRemoteGATTCharacteristic
+    private _rx_characteristic: BluetoothRemoteGATTCharacteristic
```
Comment (pelikhan) [inline src/jdom/bluetooth.ts:58]: camelized best ``_rxCharacteristic`` - i can fix later.

File: src/jdom/bluetooth.ts [hunk 2]
```diff
@@ -71,7 +74,8 @@ class BluetoothTransport extends JDTransport {
             this._device = devices?.[0]
         } else {
             const device = await bleRequestDevice({
-                filters: [{ services: [BLUETOOTH_JACDAC_SERVICE] }],
+                filters: [{namePrefix:"BBC micro:bit"}],
```
Comment (pelikhan) [inline src/jdom/bluetooth.ts:77]: filters should be an optional constructor argument to eventually support non-micro:bit BLE clients. will fix later.

File: src/jdom/bluetooth.ts [hunk 3]
```diff
@@ -91,36 +95,41 @@ class BluetoothTransport extends JDTransport {
         this._service = await this._server.getPrimaryService(
             BLUETOOTH_JACDAC_SERVICE
         )
+        console.log(`${BLUETOOTH_JACDAC_RX_CHARACTERISTIC}, ${this._service}`)
```
Comment (pelikhan) [inline src/jdom/bluetooth.ts:98]: do ``console.log({ rx: BLUETOOTH_JACDAC_RX_CHARACTERISTIC, service: this._service })``

File: src/jdom/bluetooth.ts [hunk 4]
```diff
@@ -130,7 +139,7 @@ class BluetoothTransport extends JDTransport {
             )
             this._server.disconnect()
         } finally {
-            this._characteristic = undefined
+            this._rx_characteristic = undefined
```
Comment (pelikhan) [inline src/jdom/bluetooth.ts:142]: this._tx_characteristic = undefined;

File: src/jdom/bluetooth.ts (hunk 1 above)
Comment (jamesadevine) [reply src/jdom/bluetooth.ts]: camels added!

File: src/jdom/bluetooth.ts (hunk 2 above)
Comment (jamesadevine) [reply src/jdom/bluetooth.ts:77]: TODO

File: src/jdom/bluetooth.ts (hunk 3 above)
Comment (jamesadevine) [reply src/jdom/bluetooth.ts]: removed, was debug, not required.

File: src/jdom/bluetooth.ts (hunk 4 above)
Comment (jamesadevine) [reply src/jdom/bluetooth.ts]: thanks!

File: src/jdom/bluetooth.ts (hunk 3 above)
Comment (pelikhan) [reply src/jdom/bluetooth.ts]: The object syntax is a great trick for logging

File: src/jdom/bluetooth.ts (hunk 1 above)
Comment (pelikhan) [reply src/jdom/bluetooth.ts]: Camels!

**Pull request merged**